tests/core/test_playback.py

```diff
@@ -789,3 +789,44 @@ def test(self):
         c.playback.pause()
         c.playback.next()
         b.playback.change_track.assert_called_once_with(track2)
+
+
+class Bug1352RegressionTest(unittest.TestCase):
+    def test(self):
+        config = {
+            'core': {
+                'max_tracklist_length': 10000,
+            }
+        }
+
+        b = mock.Mock()
+        b.uri_schemes.get.return_value = ['dummy']
+        b.playback = mock.Mock(spec=backend.PlaybackProvider)
+        b.playback.change_track.return_value.get.return_value = True
+        b.playback.play.return_value.get.return_value = True
+
+        track1 = Track(uri='dummy:a', length=40000)
+        track2 = Track(uri='dummy:b', length=40000)
+
+        tl_track2 = TlTrack(1, track2)
+
+        c = core.Core(config, mixer=None, backends=[b])
+        c.tracklist.add([track1, track2])
+
+        d = mock.Mock()
+        c.history._add_track = d
```

Comment:
I think the test would be more readable if you use `c.history._add_track` instead of `d` throughout the test.

Same for `c.tracklist._mark_playing` instead of `e`.